source/lac/petsc_matrix_base.cc

```diff
@@ -541,7 +541,7 @@ namespace PETScWrappers
           AssertThrow (ierr == 0, ExcPETScError(ierr));
           if (transpose_left)
             {
-              ierr = MatTranspose(tmp, MAT_REUSE_MATRIX, &tmp);
```

Comment:
@Jie-Cheng one of the testers failed to build this PR:

```
[557/859] Building CXX object source/lac/CMakeFiles/obj_lac_debug.dir/sparse_matrix.cc.o
FAILED: /usr/bin/clang++   -DDEBUG -DTBB_DO_ASSERT=1 -DTBB_IMPLEMENT_CPP0X=1 -DTBB_USE_DEBUG -Isource/lac -Iinclude -I/home/bob/source/include -I/home/bob/source/bundled/tbb41_20130401oss/include -I/home/bob/source/bundled/umfpack/UMFPACK/Include -I/home/bob/source/bundled/umfpack/AMD/Include -I/home/bob/source/bundled/boost-1.62.0/include -I/home/bob/source/bundled/muparser_v2_2_4/include -I/usr/lib/openmpi/include -I/usr/lib/openmpi/include/openmpi -I/home/bob/libs/parmetis-4.0.3/include -I/home/bob/libs/trilinos-12.4.2-Source/include -I/home/bob/libs/petsc-3.6.4/include -I/home/bob/libs/hdf5-1.8.15-patch1/include -I/home/bob/libs/oce-OCE-0.17/include/oce -I/home/bob/libs/p4est-1.1/FAST/include -I/home/bob/libs/slepc-3.6.0/include    -pedantic -fPIC -Wall -Wextra -Wpointer-arith -Wwrite-strings -Wsynth -Wsign-compare -Wswitch -Woverloaded-virtual -Wno-deprecated-declarations -Qunused-arguments -Wno-unsupported-friend -Wno-unused-parameter -Wno-unused-function -std=c++11 -Wno-parentheses -O0 -ggdb -MMD -MT source/lac/CMakeFiles/obj_lac_debug.dir/petsc_matrix_base.cc.o -MF "source/lac/CMakeFiles/obj_lac_debug.dir/petsc_matrix_base.cc.o.d" -o source/lac/CMakeFiles/obj_lac_debug.dir/petsc_matrix_base.cc.o -c /home/bob/source/source/lac/petsc_matrix_base.cc
/home/bob/source/source/lac/petsc_matrix_base.cc:544:40: error: use of undeclared identifier 'MAT_INPLACE_MATRIX'
              ierr = MatTranspose(tmp, MAT_INPLACE_MATRIX, &tmp);
                                       ^
1 error generated.
ninja: build stopped: subcommand failed.
build FAILED
```

you probably need to guard this line into something like
```
#if DEAL_II_PETSC_VERSION_LT(3,5,0)
...
#else
..
#endif
```
depending on when `MatTranspose` changed.

Reply:
Thanks. `MatTranspose` was changed in PETSc 3.8. I'll update.

Reply:
@Jie-Cheng did you have a chance to have a look at this?